packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -161,7 +161,7 @@ Directory getUserDirectory(String dirName) {
 /// These are the names of the variables in "[configHome]/user-dirs.dirs", with
 /// the `XDG_` prefix removed and the `_DIR` suffix removed.
 Set<String> getUserDirectoryNames() {
-  final File configFile = File(path.join(configHome.path, 'user-dirs.dirs'));
+  final File configFile = File(path.join(configHome!.path, 'user-dirs.dirs'));
```

Comment:
It's pretty difficult to reason about whether the `!` is safe here; it requires going and reading a bunch of other code. And if that code changed, this would become wrong.

It would be much safer to split `_directoryFromEnvironment` into a version with a fallback that can't return null and version without a fallback that can, and then many of the calls, including this one, would return non-null values.